content/en/blog/_posts/2023-01-17-confidential-kubernetes.md

```diff
@@ -0,0 +1,315 @@
+from the 2016 Usenix Security Symposium.
+
+After that, the SEV-ES feature set was implemented in
+[2020](https://www.phoronix.com/news/AMD-SEV-ES-Linux-2020-Patches). Recently, the latest feature set
+was [merged in Linux 5.19 in July 2022](https://www.phoronix.com/news/AMD-SEV-SNP-Arrives-Linux-5.19)
+and enabled support for SEV-SNP.
+
+[Confidential VMs based on AMD SEV-SNP](https://azure.microsoft.com/en-us/updates/azureconfidentialvm/)
+are available in Microsoft Azure since July 2022. Similarly, Google Cloud Platform (GCP) offers
+[confidential VMs based on AMD SEV-ES](https://cloud.google.com/compute/confidential-vm/docs/about-cvm).
+
+### Intel SGX
+
+Intel's
+[Software Guard Extensions](https://www.intel.com/content/www/us/en/developer/tools/software-guard-extensions/overview.html)
+has been available since 2015, and were introduced with the Skylake architecture.
+
+SGX is a new instruction set that enables users to create a protected and isolated process called
+an *enclave*. It provides a reverse sandbox that protects enclaves from the operating system,
+firmware, and any other privileged execution context.
+
+The enclave memory cannot be read or written from outside the enclave, regardless of
+the current privilege level and CPU mode. The only way to call an enclave function is
+through a new instruction that performs several protection checks. Its memory is encrypted.
+Tapping the memory or connecting the DRAM modules to another system will yield only encrypted
+data. The memory encryption key randomly changes every power cycle. The key is stored
+within the CPU and is not accessible.
+
+The 3rd generation Xeon CPUs (aka Ice Lake Server - "ICX") did switch to using a technology called
+[Total Memory Encryption - Multi-Key](https://www.intel.com/content/www/us/en/developer/articles/news/runtime-encryption-of-memory-with-intel-tme-mk.html)
+(TME-MK) that uses AES-XTS, moving away from the
+[Memory Encryption Engine](https://eprint.iacr.org/2016/204.pdf)
+that the consumer and Xeon E CPUs used. This increased the possible
+[enclave page cache](https://sgx101.gitbook.io/sgx101/sgx-bootstrap/enclave#enclave-page-cache-epc)
+(EPC) size (up to 512GB/CPU) as well as providing a big increase in performance. More info
+about SGX on multi-socket platforms can be found in the
+[Whitepaper](https://www.intel.com/content/dam/www/public/us/en/documents/white-papers/supporting-intel-sgx-on-mulit-socket-platforms.pdf)
+
+A [list of supported platforms](https://ark.intel.com/content/www/us/en/ark/search/featurefilter.html?productType=873&2_SoftwareGuardExtensions=Yes%20with%20both%20Intel%C2%AE%20SPS%20and%20Intel%C2%AE%20ME)
+is available from Intel.
+
+However, there are security limitations, such as cache timing attacks, physical attacks,
+untrusted I/O, or malicious microcode patching. Each project has different security requirements
+and needs to decide, if the added security of SGX and its limitations, are sufficient.
+
```

Comment:
This is the replacement. Instead of citing specific attacks we list the classes of attacks so users can use that as a starting point for more research on the topic. Do you think this section is a good starting point for that?